components/popup/src/vwc-popup-base.ts

```diff
@@ -0,0 +1,215 @@
+import {
+	PropertyValues, html, LitElement, property, query, TemplateResult
+} from 'lit-element';
+import { ClassInfo, classMap } from 'lit-html/directives/class-map.js';
+import { nothing } from 'lit-html';
+import { computePosition, offset, shift, flip, arrow } from '@floating-ui/dom';
+import type { Placement, Strategy, Padding } from '@floating-ui/core';
+
+export class VWCPopupBase extends LitElement {
+	@query('.popup-wrapper') protected popupEl!: HTMLElement;
+	@query('.popup-arrow') protected arrowEl!: HTMLElement;
+	protected padding: Padding = 0;
+	protected distance = 12;
+
+	/**
+	 * @prop open - indicates whether the popup is open
+	 * accepts boolean value
+	 * @public
+	 * */
+	@property({ type: Boolean, reflect: true })
+		open = false;
+
+	/**
+	 * @prop anchor - the anchor of the popup
+	 * accepts Element
+	 * @public
+	 * */
+	@property({ type: Element, reflect: true })
+		anchor!: Element;
+
+	/**
+	 * @prop dismissible - adds close button to the popup
+	 * accepts boolean value
+	 * @public
+	 * */
+	@property({ type: Boolean, reflect: true })
+		dismissible?: boolean;
+
+	/**
+	 * @prop corner - the placement of the popup
+	 * accepts  | 'top'
+				| 'top-start'
+				| 'top-end'
+				| 'right'
+				| 'right-start'
+				| 'right-end'
+				| 'bottom'
+				| 'bottom-start'
+				| 'bottom-end'
+				| 'left'
+				| 'left-start'
+				| 'left-end';
+	 * @public
+	 * */
+	@property({ type: String, reflect: true })
+		corner: Placement = 'left';
+
+	/**
+	 * @prop strategy - the position of the popup
+	 * accepts 'absolute' | 'fixed';
+	* @public
+	* */
+	@property({ type: String, reflect: true })
+		strategy: Strategy = 'fixed';
+
+	/**
+	 * @prop arrow - adds small triangle to indicate the trigger element
+	 * accepts boolean value
+	 * @public
+	 * */
+	@property({ type: Boolean, reflect: true })
+		arrow?: boolean;
+
+	/**
+	 * @prop alternate - set the color-scheme to dark
+	 * accepts boolean value
+	 * @public
+	 * */
+	@property({ type: Boolean, reflect: true })
+		alternate?: boolean;
+
+	/**
+	* Opens the popup
+	* @public
+	*/
+	show(): void {
+		this.open = true;
+		this.updatePosition();
+	}
+
+	/**
+	 * Closes the popup
+	 * @public
+	 */
+	hide(): void {
+		this.open = false;
+	}
+
+	override connectedCallback(): void {
+		super.connectedCallback();
+		document.addEventListener('scroll', this.updatePosition);
+		document.addEventListener('resize', this.updatePosition);
+	}
+
+	override disconnectedCallback(): void {
+		super.disconnectedCallback();
```

Comment:
@rinaok verify this doesn't cause memory leaks

Reply:
@yinonov it actually had memory leaks, I solved it by:

`private onResizeWindow = this.updatePosition.bind(this);
		document.addEventListener('scroll', this.updatePosition);
		document.removeEventListener('scroll', this.updatePosition);

`